packages/protocol/migrations_sol/integration_tests.sh

```diff
@@ -1,4 +1,66 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
-forge test --fork-url http://127.0.0.1:$ANVIL_PORT --match-contract=IntegrationTest -vvv # || echo "Test failed" # TODO for some reason the echo didn't work
\ No newline at end of file
+: '
+  The following runs specific integration tests.
+  '
+# Run integration tests
+forge test \
+--match-contract=IntegrationTest \
+-vvv \
+--fork-url http://127.0.0.1:$ANVIL_PORT
+
+: '
+  The following tests assert that existing unit tests pass when run against the migrated anvil fork.
+  These are necessary, but not sufficient requirements.
+  The idea is that existing unit tests, which are known to pass in a testing environment, should
+  also pass if they are run against the devchain using the "--fork-url" flag.
+  '
+# Run tests common
+# can't use gas limit because some setUp function use more than the limit
+forge test \
+--match-path "test-sol/common/*" \
+-vvv \
+--fork-url http://127.0.0.1:$ANVIL_PORT
+
+# Run tests governance/network
```

Comment:
```sh
# Get BlockParameters.sol address
$ cast call \
0x000000000000000000000000000000000000ce10 \
"getAddressForStringOrDie(string calldata identifier)" \
"BlockchainParameters" \
--rpc-url=http://127.0.0.1:8546
0x000000000000000000000000feb14c5787fd63a417612bf8bc2232e169a7034f

$ cast abi-decode \
"getAddressForStringOrDie(string calldata identifier)(address)" \
0x000000000000000000000000feb14c5787fd63a417612bf8bc2232e169a7034f
0xFEB14C5787fd63a417612bF8Bc2232e169A7034F
```

```sh
# Read `uptimeLookbackWindow` in BlockParameters.sol contract
$ cast call \
0xFEB14C5787fd63a417612bF8Bc2232e169A7034F \
"getUptimeLookbackWindow()(uint256)" \
--rpc-url=http://127.0.0.1:8546
Error:
(code: 3, message: execution reverted: UptimeLookbackWindow is not initialized, data: Some(String("0x08c379a000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000027557074696d654c6f6f6b6261636b57696e646f77206973206e6f7420696e697469616c697a656400000000000000000000000000000000000000000000000000")))
```

The read call fails, which indicates that the contract is not intialized correctly.

This error is thrown here:

https://github.com/celo-org/celo-monorepo/blob/fc1e4e3559689482ddf0b9647ccb383ad7627e32/packages/protocol/contracts/governance/BlockchainParameters.sol#L114-L117

Reply:
My current thinking is that the contract initialization started here fails:

https://github.com/celo-org/celo-monorepo/blob/cd987f334c9422e5e4a817b68a853efa7200aa7f/packages/protocol/contracts/governance/BlockchainParameters.sol#L51-L61

It calls this function 

https://github.com/celo-org/celo-monorepo/blob/cd987f334c9422e5e4a817b68a853efa7200aa7f/packages/protocol/contracts/governance/BlockchainParameters.sol#L95-L109

I have a hunch the problem has to do with this `require` statement:

https://github.com/celo-org/celo-monorepo/blob/cd987f334c9422e5e4a817b68a853efa7200aa7f/packages/protocol/contracts/governance/BlockchainParameters.sol#L97-L101

It uses this function defined in: 

https://github.com/celo-org/celo-monorepo/blob/cd987f334c9422e5e4a817b68a853efa7200aa7f/packages/protocol/contracts/common/UsingPrecompiles.sol#L57-L64

Reply:
I ran the migration script with more verbose logging to find out what happens during the initialization:

```sh
  [188746] Proxy::_setAndInitializeImplementation(BlockchainParameters: [0xd7e23B7A53b25dFAef176Cf96601767C999a2633], 0x80d85911000000000000000000000000000000000000000000000000000000000000c3500000000000000000000000000000000000000000000000000000000000c65d40000000000000000000000000000000000000000000000000000000000000000c)
    ├─ emit ImplementationSet(implementation: BlockchainParameters: [0xd7e23B7A53b25dFAef176Cf96601767C999a2633])
    ├─ [125974] BlockchainParameters::initialize(50000 [5e4], 13000000 [1.3e7], 12) [delegatecall]
    │   ├─ emit OwnershipTransferred(previousOwner: 0x0000000000000000000000000000000000000000, newOwner: 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266)
    │   ├─ emit BlockGasLimitSet(limit: 13000000 [1.3e7])
    │   ├─ emit IntrinsicGasForAlternativeFeeCurrencySet(gas: 50000 [5e4])
    │   ├─ [181] 0x00000000000000000000000000000000000000f8::fallback(0x01) [staticcall]
    │   │   └─ ← 0x0000000000000000000000000000000000000000000000000000000000004380
    │   ├─ [181] 0x00000000000000000000000000000000000000f8::fallback(0x01) [staticcall]
    │   │   └─ ← 0x0000000000000000000000000000000000000000000000000000000000004380
    │   ├─ [181] 0x00000000000000000000000000000000000000f8::fallback(0x01) [staticcall]
    │   │   └─ ← 0x0000000000000000000000000000000000000000000000000000000000004380
    │   ├─ emit UptimeLookbackWindowSet(window: 12, activationEpoch: 2)
    │   └─ ← ()
    └─ ← ()
```

Reply:
Looks like it's using `getUptimeLookbackWindow`:

https://github.com/celo-org/celo-monorepo/blob/cd987f334c9422e5e4a817b68a853efa7200aa7f/packages/protocol/contracts/governance/BlockchainParameters.sol#L114-L117

which is calling:

https://github.com/celo-org/celo-monorepo/blob/cd987f334c9422e5e4a817b68a853efa7200aa7f/packages/protocol/contracts/governance/BlockchainParameters.sol#L122-L127

which is calling:

https://github.com/celo-org/celo-monorepo/blob/cd987f334c9422e5e4a817b68a853efa7200aa7f/packages/protocol/contracts/common/UsingPrecompiles.sol#L78-L80

which is calling 

https://github.com/celo-org/celo-monorepo/blob/cd987f334c9422e5e4a817b68a853efa7200aa7f/packages/protocol/contracts/common/UsingPrecompiles.sol#L70-L72

which is calling

https://github.com/celo-org/celo-monorepo/blob/cd987f334c9422e5e4a817b68a853efa7200aa7f/packages/protocol/contracts/common/UsingPrecompiles.sol#L232-L243

Reply:
From these more verbose test logs, we know that the test fails because:

```diff
+ Expected: emit UptimeLookbackWindowSet(window: 20, activationEpoch: 2)
- Actual: emit UptimeLookbackWindowSet(window: 20, activationEpoch: 4)
```

Reply:
This PR only focuses on integration tests and not end-to-end tests, so this failing test is out of scope.